### PR TITLE
introduce record specific enabling/disabling of actions in Grid/CRUD

### PR DESCRIPTION
**Pull request by georgehristov**
Enabling the Actions column decorators to make use of `$action->enabled` attribute when it is has a callable value. The `$row` Model is passed on as parameter allowing to define if action is enabled or disabled for each row in the grid individually.

Code was also minor refactored to remove duplication.

Attributes and methods were renamed to be more expressive and represent actual functionality.

### CODE REVIEW

File: src/TableColumn/ActionButtons.php [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        }
+        $this->table->on('click', '.b_'.$name, $action, [$this->table->jsRow()->data('id'), 'confirm' => $confirm]);
+
+        return $button;
+    }
+
+    /**
+     * Adds a new button which will open a modal dialog and dynamically
+     * load contents through $callback. Will pass a virtual page.
+     */
```
Comment (DarkSide666) [inline src/TableColumn/ActionButtons.php]: please add @param section

File: src/TableColumn/ActionButtons.php [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+use atk4\core\FactoryTrait;
+
+/**
+ * Formatting action buttons column.
+ */
+class ActionButtons extends Generic
+{
+    use FactoryTrait;
+
+    public $buttons = [];
```
Comment (DarkSide666) [inline src/TableColumn/ActionButtons.php]: please add comment section

**Comment by ibelar** [comment]
This look very good. Good refactoring too. Please fix conflicting file and we can merge.

**Comment by romaninsh** [comment]
@georgehristov can you please update / resolve conflicts? I like the idea!

**Comment by georgehristov** [comment]
Merging conflicts resolved

**Comment by georgehristov** [comment]
This PR stalled. Shall I do something more so it gets merged?

**Comment by romaninsh** [comment]
I'm happy to merge. May have some performance implications though.

**Pull request merged**